azurerm/internal/services/timeseriesinsights/resource_time_series_insights_environment.go

```diff
@@ -0,0 +1,293 @@
+	defer cancel()
+
+	name := d.Get("name").(string)
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	resourceGroup := d.Get("resource_group_name").(string)
+	t := d.Get("tags").(map[string]interface{})
+	sku, err := expandEnvironmentSkuName(d.Get("sku_name").(string))
+	if err != nil {
+		return fmt.Errorf("expanding sku: %+v", err)
+	}
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, name, "")
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for presence of existing Time Series Insights Environment %q (Resource Group %q): %s", name, resourceGroup, err)
+			}
+		}
+
+		if existing.Value != nil {
+			environment, ok := existing.Value.AsStandardEnvironmentResource()
+			if !ok {
+				return fmt.Errorf("exisiting resource was not a standard Time Series Insights Environment %q (Resource Group %q)", name, resourceGroup)
+			}
+
+			if environment.ID != nil && *environment.ID != "" {
+				return tf.ImportAsExistsError("azurerm_time_series_insights_environment", *environment.ID)
+			}
+		}
+	}
+
+	environment := timeseriesinsights.StandardEnvironmentCreateOrUpdateParameters{
+		Location: &location,
+		Tags:     tags.Expand(t),
+		Sku:      sku,
+		StandardEnvironmentCreationProperties: &timeseriesinsights.StandardEnvironmentCreationProperties{
+			StorageLimitExceededBehavior: timeseriesinsights.StorageLimitExceededBehavior(d.Get("storage_limited_exceeded_behavior").(string)),
+			DataRetentionTime:            utils.String(d.Get("data_retention_time").(string)),
+		},
+	}
+
+	if v, ok := d.GetOk("partition_key"); ok {
+		partition := make([]timeseriesinsights.TimeSeriesIDProperty, 1)
+		partition[0] = timeseriesinsights.TimeSeriesIDProperty{
+			Name: utils.String(v.(string)),
+			Type: timeseriesinsights.String,
+		}
+		environment.StandardEnvironmentCreationProperties.PartitionKeyProperties = &partition
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, name, environment)
+	if err != nil {
+		return fmt.Errorf("creating/updating Time Series Insights Environment %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for completion of Time Series Insights Environment %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, name, "")
+	if err != nil {
+		return fmt.Errorf("retrieving Time Series Insights Environment %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	resource, ok := resp.Value.AsStandardEnvironmentResource()
+	if !ok {
+		return fmt.Errorf("resource was not a standard Time Series Insights Environment %q (Resource Group %q)", name, resourceGroup)
+	}
+
+	if resource.ID == nil {
+		return fmt.Errorf("cannot read Time Series Insights Environment %q (Resource Group %q) ID", name, resourceGroup)
+	}
```

Comment:
Could we add a check to ensure the ID is not nil and not empty?
```suggestion
	if resource.ID == nil || *resource.ID == "" {
		return fmt.Errorf("cannot read Time Series Insights Environment %q (Resource Group %q) ID", name, resourceGroup)
	}
```